File: javascript-closure-angular-client/API/Client/TaxTypeRate.js
goog.provide('API.Client.TaxTypeRate');

/**
 * @record
 */
API.Client.TaxTypeRate = function() {}

/**
 * @type {!API.Client.TaxType}
 * @export
 */
API.Client.TaxTypeRate.prototype.taxType;

/**
 * Tax can be calculated by rate or by quantity
 * @type {!string}
 * @export
 */
API.Client.TaxTypeRate.prototype.taxModel;

/**
 * Specific tax rate ex 3.5 (3.5%)
 * @type {!number}
 * @export
 */
API.Client.TaxTypeRate.prototype.rate;

/**
 * Specific tax rate ex 3.5 (3.5%)
 * @type {!number}
 * @export
 */
API.Client.TaxTypeRate.prototype.srvAmount;

/**
 * @type {!string}
 * @export
 */
API.Client.TaxTypeRate.prototype.quantityUnidBase;

/**
 * @type {!string}
 * @export
 */
API.Client.TaxTypeRate.prototype.specializationType;

/** @enum {string} */
API.Client.TaxTypeRate.TaxModelEnum = { 
  rate: 'rate',
  srf: 'srf',
}
/** @enum {string} */
API.Client.TaxTypeRate.SpecializationTypeEnum = { 
  basic: 'basic',
  monophase: 'monophase',
  taxSubstitution: 'taxSubstitution',
}

File: async-scala-client/src/main/scala/io/swagger/client/model/TaxTypeRate.scala
package io.swagger.client.model

import org.joda.time.DateTime
import java.util.UUID


case class TaxTypeRate (
  taxType: TaxType,
taxModel: String,  // Tax can be calculated by rate or by quantity
rate: Option[Double],  // Specific tax rate ex 3.5 (3.5%)
srvAmount: Option[Double],  // Specific tax rate ex 3.5 (3.5%)
quantityUnidBase: Option[String],
specializationType: Option[String],
exemptionReasonCode: Option[String],  // UUID Reference to an item in the LegalReason store. 
customExemptionReasonDescription: Option[String],  // Optional textual reason description, to be used when reason codes are generic (i.e. reason code 999 &#x3D; Other). 
withholding: Option[TaxTypeRate_withholding]
)
